packages/db/src/paginate/find.js

```diff
@@ -59,6 +60,13 @@ module.exports = async (basedb, modelName, {
   if (collate) options.collation = $sort.collation;
 
   const results = await basedb.find(modelName, $query, options);
-
+  if (isArray(ids) && ids.length) {
+    const sorted = ids
+      // Resort the items to match the ID order
+      .map(id => results.find(({ _id }) => `${_id}` === `${id}`))
```

Comment:
Convert results into `Map` object, keyed by the sort field. Then run `ids.map` and "pluck" the value out of the map.